File: macroslib/tests/expectations/return_result_with_object_as_value_and_err.rs
foreigner_class!(class Position {
    self_type GnssInfo;
    private constructor create_position() -> GnssInfo;
    fn Position::getLatitude(&self) -> f64;
});

foreigner_class!(class LocationService {
    self_type LocationService;

    constructor LocationService::new() -> LocationService;
    fn LocationService::f1() -> Result<GnssInfo, String>;
    fn LocationService::f2() -> Result<(), String>;

    fn LocationService::create() -> Result<LocationService, String>;
});

foreigner_class!(class Foo {
    self_type Foo<'a>;
    constructor Foo::create() -> Foo<'a>;
    fn Foo::from_string<'a>(_: &str) -> Result<Foo<'a>, String>;
});

File: macroslib/tests/expectations/return_result_with_object_as_value_and_err.java
r#"public static @NonNull Position f1() throws Exception {
        long ret = do_f1();
        Position convRet = new Position(InternalPointerMarker.RAW_PTR, ret);

        return convRet;
    }
    private static native long do_f1() throws Exception;"#;
"public static native void f2() throws Exception;";

r#"public static @NonNull LocationService create() throws Exception {
        long ret = do_create();
        LocationService convRet = new LocationService(InternalPointerMarker.RAW_PTR, ret);

        return convRet;
    }
    private static native long do_create() throws Exception;"#;

r#"public static @NonNull Foo from_string(@NonNull String a0) throws Exception {
        long ret = do_from_string(a0);
        Foo convRet = new Foo(InternalPointerMarker.RAW_PTR, ret);

        return convRet;
    }
    private static native long do_from_string(@NonNull String a0) throws Exception;"#;
